fluid/utils.py

```diff
@@ -12,12 +12,59 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from compiler.ast import flatten
 
-def get_op_io_info(op):
-    inputs = dict([(name, op.input(name)) for name in op.input_names])
-    attrs = dict(
-        [(name, op.attr(name))
-         for name in op.attr_names]) if op.attr_names is not None else None
-    outputs = dict([(name, op.output(name)) for name in op.output_names])
 
-    return inputs, attrs, outputs
+class UniqOpIOs():
+    """Return input/output information for an operator, and resolve potential 
```

Comment:
In Fluid `Program`, e.g. activations, input and output may share the same name to optimize memory  utility. Changes here are going to solve this problem.

Reply:
Nice find, I am glad you looked into this because this conflict would be pretty annoying

Reply:
Yes. Also, we may need to rename the names in fetch_list in case that these memory-shared variables are fetched.

Reply:
The new update has considered the rename of graph outputs.